src/main/java/frameworksdrivers/Gateways/SessionDbGateway.java

```diff
@@ -232,22 +234,20 @@ public EvaluatorDbResponseModel retrieveCourse() {
     /**
      * Retrieves the courseID of the current user logged in. There can only be one user in the session database thus
      * it returns the only username on the database.
-     *
-     * @return String being the username of the current student.
      */
     @Override
-    public String retrieveStudentId() {
+    public void retrieveUser(EvaluatorDbRequestModel requestModel) {
         try {
-            String SQL = "SELECT DISTINCT StudentID FROM " + DATABASE_NAME_STUDENT;
+            String SQL = "SELECT DISTINCT StudentID, Password FROM " + DATABASE_NAME_STUDENT;
             PreparedStatement statement = connection.prepareStatement(SQL);
             ResultSet resultSet = statement.executeQuery();
 
             resultSet.next();
-            return resultSet.getString(1);
+            requestModel.setStudentID(resultSet.getString(1));
+            requestModel.setPassword(resultSet.getString(2));
```

Comment:
I changed this method because now the Evaluator must retrieve the entire user (the username and password) from the session database. Not only the student id.

Reply:
So does this method also check if the course has been saved with the student's username because that could be the case if they're retaking the course?